Completed fourth part of Diploma.

VlasovM opened the pull request:
Added own annotation to validate Email, password and name, when user register.
Added Spring security: view your posts, login, logout.

Commit:
Added own annotations validator.

Validate email, name and password in Register response.

Commit:
Merge branch 'master' into dev

Commit:
Completed third part and fixed some methods.

Added corrected view count in posts (when we look at our posts). Added login, logout and see my posts API with Spring Security. Renamed Dto for posts and created Dto for User. (UserDTO and UserDtoForPosts). Renamed package entity to model and created package enums.

Commit:
Fixed small bugs and replaced values to Optional in request param.

Klepus commented on pom.xml:
```diff
@@ -54,13 +49,33 @@
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+            <version>6.2.0.Final</version>
```
версия

Klepus commented on src/main/java/com/javlasov/blog/controller/ApiGeneralController.java:
```diff
@@ -38,13 +38,12 @@ private ResponseEntity<SettingsResponse> settings() {
-    private ResponseEntity<CalendarResponse> calendar(@RequestParam(required = false, defaultValue = "0") int year) {
-        return (year == 0) ? ResponseEntity.ok(calendarService.calendar(LocalDate.now().getYear()))
-                : ResponseEntity.ok(calendarService.calendar(year));
+    private ResponseEntity<CalendarResponse> calendar(@RequestParam(required = false) Optional<Integer> year) {
```
private?

Klepus commented on src/main/java/com/javlasov/blog/controller/ApiGeneralController.java:
```diff
@@ -38,13 +38,12 @@ private ResponseEntity<SettingsResponse> settings() {
 
     @GetMapping("/tag")
-    private ResponseEntity<TagResponse> tag(@RequestParam(required = false, defaultValue = "") String query) {
+    private ResponseEntity<TagResponse> tag(@RequestParam(required = false) Optional<String> query) {
```
почему private? как тестировать потом будешь?

Klepus commented on src/main/java/com/javlasov/blog/dto/PostCommentDto.java:
```diff
@@ -10,6 +10,6 @@ public class PostCommentDto {
     private long timestamp;
     private String text;
-    private UserDto user;
+    private UserDtoForPosts user;
```
мб UserPostsDto?

Commit:
Fixed bugs and corrected code style.

Replace controllers to public, fixed getting auth, when user not found, added margins in classes for better readability.
